apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/utils/TradingViewUtils.java

```diff
@@ -171,15 +171,12 @@ public static TradingDataOutput getDataForIntervalFromOffers( String fsym, Strin
             if (dexOrdersForInterval.size() > 0 && log.isTraceEnabled()) {
                 log.trace ("interval data added, i: {} ts: {}, lo: {}, hi: {}, open: {}, close : {}", i, entryForPeriod.time, entryForPeriod.low, entryForPeriod.high, entryForPeriod.open, entryForPeriod.close);
             }
-            initialTime += interval;                
+            initialTime += interval;   
             
-            if (!entryForPeriod.isZero()) {                
-                data.add(entryForPeriod);
-                } else {                
-                if (i==limit-1) {                    
-                    data.add(entryForPeriod);
-                    }
-                }
+            entryForPeriod.open = prevClose;
+            prevClose = entryForPeriod.getClose();
```

Comment:
When current entry is zero, this code will set prevClose to ZERO and next zero/non-zero candlesticks will have zero open price.